MAUI/DataForm/DataForm-settings.md

```diff
@@ -0,0 +1,899 @@
+</td>
+</tr>
+<tr>
+<td>
+DataFormTimeItem
+</td>
+<td>
+{{'[TimePicker](https://learn.microsoft.com/en-us/dotnet/maui/user-interface/controls/timepicker?view=net-maui-7.0&tabs=windows)'| markdownify }}
+</td>
+<td>
+Generated for TimeSpan or TimeOnly property type and properties with [DataType(DataType.Time)] attribute. 
+</td>
+</tr>
+<tr>
+<td>
+DataFormPickerItem
+</td>
+<td>
+{{'[Picker](https://learn.microsoft.com/en-us/dotnet/maui/user-interface/controls/picker?view=net-maui-7.0)'| markdownify }}
+</td>
+<td>
+Generated for Enum type property. 
+</td>
+</tr>
+<tr>
+<td>
+DataFormCheckBoxItem
+</td>
+<td>
+{{'[CheckBox](https://learn.microsoft.com/en-us/dotnet/maui/user-interface/controls/checkbox?view=net-maui-7.0)'| markdownify }}
+</td>
+<td>
+Generated for the Bool type property.
+</td>
+</tr>
+<tr>
+<td>
+DataFormSwitchItem
+</td>
+<td>
+{{'[Switch](https://learn.microsoft.com/en-us/dotnet/maui/user-interface/controls/switch?view=net-maui-7.0)'| markdownify }}
+</td>
+<td>
+Generated for the Bool type property.
+</td>
+</tr>
+<tr>
+<td>
+DataFormComboBoxItem
+</td>
+<td>
+{{'[SfComboBox](https://help.syncfusion.com/maui/combobox)'| markdownify }}
+</td>
+<td>
+Generated for Enum type property. 
+</td>
+</tr><tr>
+<td>
+`DataFormAutoCompleteItem`
+</td>
+<td>
+{{'[SfAutoComplete](https://help.syncfusion.com/maui/autocomplete)'| markdownify }}
+</td>
+<td>
+Generated for Enum type property. 
+</td>
+</tr>
+<tr>
+<td>
+DataFormRadioGroupItem
+</td>
+<td>
+{{'[RadioButton](https://learn.microsoft.com/en-us/dotnet/maui/user-interface/controls/radiobutton?view=net-maui-7.0)'| markdownify }}
+</td>
+<td>
+Generated for the Enum type property.
+</td>
+</tr>
+</table>
+
+You can customize the property settings or cancel the generation of `DataFormItem` by handling the `GenerateDataFormItem` event.
+
+## Cancel to generate data editor
+
+You can cancel the specific data editor by handling the `GenerateDataFormItem` event or by defining display attribute to avoid the particular data field being displayed.
+
+#### Using attributes
+
+You can set `AutoGenerateField` to `false` for canceling the data editor generation.
+
+{% tabs %}
+{% highlight C# %}
+
+    [Display(AutoGenerateField = false)]
+    public int ID { get; set; } 
+    
+{% endhighlight %}
+{% endtabs %}
+
+#### Using event
+
+In the following code, the data editor generation for the `MiddleName` property is canceled by setting the `Cancel` property to true of the `DataFormItem`.
+
+{% tabs %}
+{% highlight C# %}
+
+    dataForm.GenerateDataFormItem += OnGenerateDataFormItem;
+
+    private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+    {
+        if (e.DataFormItem.FieldName == "MiddleName")
+        {
+            e.Cancel = true;
+
+        }
+    }
+
+{% endhighlight %}
+{% endtabs %}
+
+GenerateDataFormItem event when AutoGenerate is true.
+
```

Comment:
For `AutoGenerateItems` false you can use settings properties directly by using the `DataFormItem`. Please refer here to know more about creating data editors explicitly.

Reply:
content updated